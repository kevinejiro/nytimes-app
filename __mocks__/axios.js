const axiosMock = jest.genMockFromModule('axios');

let mockResponse = {
    data: {
        "articles" :
        [{
            "something": "test something",
            "other" : "test other"
        }]
    }
}

axiosMock.get.mockImplementation(req);

function req() {
    return new Promise(function(resolve) {
        axiosMock.delayTimer = setTimeout(function() {
            resolve(mockResponse);
        }, 3000);
    })
}

export default axiosMock;
