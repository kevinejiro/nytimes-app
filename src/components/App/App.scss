.footer {
	background: linear-gradient(0deg, #707070 0%, #eeeeee 75%);
  text-align: center;
  position: absolute;
  bottom: 0;
  width: 100%;
  /* Set the fixed height of the footer here */
	height: 60px;
}

a {
	
	&:hover {
		color: #903a3a;
			text-decoration: none;
	}
}

.navbarbox  {
  height: 90px;
  white-space: nowrap;
  overflow-x: auto;
	background-color: #222222;
	position: relative;
}

.navbar-brand {
	float: left;

  img {
    height: 60px;
		width: 400px;
		align-self: center;
		align-items: center;
		align-content: center;
  }

  @media only screen and (max-width: 768px) {
    /* For mobile phones: */
    img {
      width: 250px;
			height: 40px;
    }
  }
}

.spinner-border {
  margin-top: 60%;
  margin-bottom: 50%;
}

.fa-chevron-left {
  align-self: center;
	padding-right: 0.1em;
	padding-left: 0.5em;
  color: #eee;
}
