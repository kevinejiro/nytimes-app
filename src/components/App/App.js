import React, { Component } from 'react';
import axios from 'axios';
import { Route } from 'react-router-dom';
import { Spinner } from 'reactstrap';

import './App.scss';
import { AppHeader } from '../common/AppHeader.js'
import { AppFooter } from '../common/AppFooter.js';

import { ArticleList } from '../ArticleList/ArticleList.js';
import { ArticleDetails } from '../ArticleDetails/ArticleDetails.js';

class App extends Component {

  state = { articles: [], isLoading: false }

  async componentDidMount() {
    this.setState({ isLoading: true });
    try {
      const response = await axios.get('https://api.nytimes.com/svc/mostpopular/v2/mostviewed/all-sections/7.json?api-key=MJ4FfwRcFW2UD0b72IBaFxA0aFMnRafp')
      this.setState({ articles: response.data.results, isLoading: false });

    } catch (error) {
      this.setState({ articles: [], isLoading: false });
    }
  }

  render() {
    const { articles, isLoading } = this.state;

    return (
      <div className="App">
        <AppHeader />
        {isLoading ? (<div className="text-center spinner-dark">
          <Spinner color="dark" />
        </div>) :
          [
            <Route
              key="article-list-router"
              exact
              path='/'
              render={() => <ArticleList articles={articles} />}
            />,
            <Route
              key="article-router"
              exact
              path='/:articleId'
              render={(matchProps) => <ArticleDetails articles={articles} {...matchProps} />}
            />
          ]
        }
        <AppFooter />
      </div>
    );
  }
}

export default App;
