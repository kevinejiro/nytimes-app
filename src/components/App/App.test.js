import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios';

import App from './App';


describe('App', () => {
	let wrapper;
	let setStateSpy;
	beforeEach(() => {
		setStateSpy = jest.spyOn(App.prototype, 'setState');
		wrapper = shallow(<App />);
	});

	it('renders without crashing', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('calls axios.get in componentDidMount', () => {
		return wrapper.instance().componentDidMount().then(() => {
			expect(axios.get).toHaveBeenCalled();
		})
	});

	it('calls setState in componentDidMount', () => {
		return wrapper.instance().componentDidMount().then(() => {
			expect(setStateSpy).toHaveBeenCalled();
		})
	});

	it('calls axios.get with correct url', () => {
		return wrapper.instance().componentDidMount().then(() => {
			expect(axios.get).toHaveBeenCalledWith('https://api.nytimes.com/svc/mostpopular/v2/mostviewed/all-sections/7.json?api-key=MJ4FfwRcFW2UD0b72IBaFxA0aFMnRafp');
		})
	});

	it('updates the state when componentDidMount is called', () => {
		return wrapper.instance().componentDidMount().then(() => {
			expect(wrapper.state()).toHaveProperty('isLoading', false
			);
		});
	});

});