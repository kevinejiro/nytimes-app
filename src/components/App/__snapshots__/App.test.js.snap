// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(undefined) />,
        <div
          className="text-center spinner-dark"
        >
          <Spinner
            color="dark"
            tag="div"
            type="border"
          >
            Loading...
          </Spinner>
        </div>,
        <UNDEFINED />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Spinner
            color="dark"
            tag="div"
            type="border"
          >
            Loading...
          </Spinner>,
          "className": "text-center spinner-dark",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Loading...",
            "color": "dark",
            "tag": "div",
            "type": "border",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(undefined) />,
          <div
            className="text-center spinner-dark"
          >
            <Spinner
              color="dark"
              tag="div"
              type="border"
            >
              Loading...
            </Spinner>
          </div>,
          <UNDEFINED />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Spinner
              color="dark"
              tag="div"
              type="border"
            >
              Loading...
            </Spinner>,
            "className": "text-center spinner-dark",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Loading...",
              "color": "dark",
              "tag": "div",
              "type": "border",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "type": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
